cmd/calyptia/update_core_instance_k8s.go

```diff
@@ -0,0 +1,144 @@
+
+func newCmdUpdateCoreInstanceK8s(config *config, testClientSet kubernetes.Interface) *cobra.Command {
+	var newVersion, newName string
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	cmd := &cobra.Command{
+		Use:               "kubernetes CORE_INSTANCE",
+		Aliases:           []string{"kube", "k8s"},
+		Short:             "update a core instance from kubernetes",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: config.completeAggregators,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			aggregatorKey := args[0]
+
+			err := VerifyCoreVersion(newVersion)
+			if err != nil {
+				return err
+			}
+
+			aggregatorID, err := config.loadAggregatorID(aggregatorKey)
+			if aggregatorKey == newName {
+				return fmt.Errorf("cannot update core instance with the same name")
+			}
+			if newName != "" {
+				err = config.cloud.UpdateAggregator(config.ctx, aggregatorID, cloud.UpdateAggregator{
+					Name: &newName,
+				})
+				if err != nil {
+					return fmt.Errorf("could not update core instance: %w", err)
+				}
+			}
+			if err != nil {
+				return err
+			}
+			agg, err := config.cloud.Aggregator(ctx, aggregatorID)
+			if err != nil {
+				return err
+			}
+
+			if configOverrides.Context.Namespace == "" {
+				configOverrides.Context.Namespace = apiv1.NamespaceDefault
+			}
+
+			var clientset kubernetes.Interface
+			if testClientSet != nil {
+				clientset = testClientSet
+			} else {
+				kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+				kubeClientConfig, err := kubeConfig.ClientConfig()
+				if err != nil {
+					return err
+				}
+
+				clientset, err = kubernetes.NewForConfig(kubeClientConfig)
+				if err != nil {
+					return err
+				}
+
+			}
+
+			k8sClient := &k8s.Client{
+				Interface:    clientset,
+				Namespace:    configOverrides.Context.Namespace,
+				ProjectToken: config.projectToken,
+				CloudBaseURL: config.baseURL,
+			}
+
+			if err := k8sClient.EnsureOwnNamespace(ctx); err != nil {
+				return fmt.Errorf("could not ensure kubernetes namespace exists: %w", err)
+			}
+
+			deploymentName := agg.Name + "-deployment"
+			if err := k8sClient.UpdateDeployment(ctx, deploymentName, coreDockerImage, newVersion); err != nil {
+				return fmt.Errorf("could not update deployment %s: %w", deploymentName, err)
+			}
+
+			if err != nil {
+				return err
+			}
+			cmd.Printf("calyptia-core instance %q was successfully updated to version %s\n", agg.Name, newVersion)
+			return nil
+		},
+	}
+	fs := cmd.Flags()
+	fs.StringVar(&newVersion, "version", "", "New version of the calyptia-core instance")
+	fs.StringVar(&newName, "name", "", "New core instance name")
+	clientcmd.BindOverrideFlags(configOverrides, fs, clientcmd.RecommendedConfigOverrideFlags("kube-"))
+	return cmd
+}
+
+func VerifyCoreVersion(newVersion string) error {
```

Comment:
verifyCoreImageVersion -- Add unit tests for this function (as is exported). (cover the 2 cases - good/bad request to the index).